contrib/initramfs/Makefile.am

```diff
@@ -6,10 +6,15 @@ initrd_SCRIPTS = \
 SUBDIRS = hooks scripts
 
 EXTRA_DIST = \
+	$(top_srcdir)/etc/init.d/zfs \
+	$(top_srcdir)/etc/init.d/zfs-functions \
 	$(top_srcdir)/contrib/initramfs/conf.d/zfs \
 	$(top_srcdir)/contrib/initramfs/conf-hooks.d/zfs \
 	$(top_srcdir)/contrib/initramfs/README.initramfs.markdown
 
+$(top_srcdir)/etc/init.d/zfs $(top_srcdir)/etc/init.d/zfs-functions:
```

Comment:
ok, this is what i was lacking in my change.  

is this really the right way to do it?

Reply:
Well, normally you would use Make dependencies here... that's what you tried to do with `SUBDIRS`. The problem here is, that AFAIK in Make one cannot specify a dependency that is one (or more) make level(s) up. This here is more like a "hack".

Reply:
I could also have specified the dependency in the top level like this:

/Makefile.am

~~~
$(top_srcdir)/contrib/initramfs: $(top_srcdir)/etc/init.d/zfs $(top_srcdir)/etc/init.d/zfs-functions
~~~

This would not handle the case of doing a `make -C contrib/initramfs`, though.

Reply:
yeah, so it's putting zfs-functions in the zfs.deb at present, that seems reasonable

i didn't entirely look at all the changes but i can verify it *does* build on my local pipeline here though i didn't do an install/boot test and won't be able to

i wonder if we can have all these commits merged together, that way it would be easier to cherry-pick for 0.8.x?

Reply:
The commits will be squashed anyways, either be me or the maintainer who merges it. I‘ll keep it this way to make review easier